### PR TITLE
Circle CI Testing

### PR DESCRIPTION
**Pull request by Xorima**
This commit adds circle ci integration testing for all tests

Signed-off-by: Jason Field <jason@avon-lea.co.uk>

### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/openvpn/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,3 +11,93 @@ workflows:
...
+          suite: server
+          platform: debian-9
+          requires: [  danger, lint ]
+      - kitchen/dokken-single:
+          name: server-centos-7
+          suite: server
+          platform: centos-7
+          requires: [  danger, lint ]
+      - kitchen/dokken-single:
+          name: server-fedora-latest
```
Comment (damacus) [inline .circleci/config.yml]: No fedora please ☺

Comment (damacus) [reply .circleci/config.yml]: You might be better off using the dokken multi.if that doesn't fit your needs we need to change that

File: .kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+# currently only support 2 last major revs of distros (at the most)
+platforms:
+  - name: centos-6
+  - name: centos-7
+  - name: debian-8
+  - name: debian-9
+  - name: ubuntu-16.04
+  - name: ubuntu-18.04
+  - name: fedora-latest
```
Comment (damacus) [inline .kitchen.yml]: Remove me and add amazon and amazon 2

File: .kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+suites:
+  - name: server
+    run_list:
+      - recipe[openvpn::server]
+    attributes:
```
Comment (damacus) [inline .kitchen.yml]: I would ideally like this to either not exist or be in the default attributes

File: .circleci/config.yml (hunk 1 above)
Comment (Xorima) [reply .circleci/config.yml]: Are we removing fedora from dokken-multi? sorry haven't been keeping up to date

Comment (damacus) [reply .circleci/config.yml]: The newst one does. @tas50 did a survey of Chef users and most didn't user Fedora, and it was  the distro that gave us the most greif. 

That and Fedora bugs aren't fixed until they're in RHEL quite often :(

**Comment by jeffbyrnes** [comment]
Ugh, sorry sorry sorry, did _not_ mean to do that.

**Comment by jeffbyrnes** [comment]
Ok, to explain: seemed like it’d be nice to have the integration tests running as well, and this PR looked like it was _really_ close, so I tightened things up, dropped the modifications to the attribute defaults in favor of what is in `.kitchen.yml` on master, brought back testing on fedora, and opened a new branch.

Sadly, I forgot that my `rename-branch` alias does some way-too-fancy pushing to the remote, and thus is blew up the branch here.

Everything’s been restored, and this is re-opened b/c I’m not keen to go off & close people’s PRs like that.

Please see #143 for my rebuild of this PR, which may get the integration tests working on CircleCI (we’ll see).

**Comment by damacus** [comment]
Closing in favour of #143

**Pull request closed without merging**